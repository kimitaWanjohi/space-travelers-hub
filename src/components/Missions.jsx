import { useSelector, useDispatch } from "react-redux";
import { useEffect } from "react";
import { fetchMissions } from "../redux/missions/slice";
import Mission from "./Mission";

export default function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missions);
  const loading = useSelector((state) => state.missions.loading);
  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);
  return (
    <div className="container mx-auto py-8 w-full">
      {loading ? (
        <h1>Loading...</h1>
      ) : (
        <table className="table-auto max-w-[90%] mx-auto bg-white shadow-md rounded-lg">
          <thead className="bg-gray-100">
            <tr>
              <th className="px-4 py-2">Mission</th>
              <th className="px-4 py-2">Description</th>
              <th className="px-4 py-2">Status</th>
              <th className="px-4 py-2">Action</th>
            </tr>
          </thead>
          <tbody>
            {missions.map((mission) => (
              <Mission key={mission.mission_id} mission={mission} />
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
}
