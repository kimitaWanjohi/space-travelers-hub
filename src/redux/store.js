import { configureStore } from "@reduxjs/toolkit";
import { createLogger } from "redux-logger";
import missionsReducer from "./missions/slice";

export default configureStore({
  reducer: {
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(createLogger()),
});
