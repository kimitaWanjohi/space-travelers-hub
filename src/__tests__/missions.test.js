import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from '../redux/store';
import Missions from "../components/Missions";

describe("Missions", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders a list of missions", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>
      )
      .toJSON();
    expect(tree.children.length).toBe(1);
  });
});
