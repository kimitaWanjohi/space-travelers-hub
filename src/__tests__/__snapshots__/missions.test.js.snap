// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
<div
  className="container mx-auto py-8 w-full"
>
  <table
    className="table-auto max-w-[90%] mx-auto bg-white shadow-md rounded-lg"
  >
    <thead
      className="bg-gray-100"
    >
      <tr>
        <th
          className="px-4 py-2"
        >
          Mission
        </th>
        <th
          className="px-4 py-2"
        >
          Description
        </th>
        <th
          className="px-4 py-2"
        >
          Status
        </th>
        <th
          className="px-4 py-2"
        >
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
